features/frameworks/utest/source/utest_harness.cpp

```diff
@@ -47,8 +47,15 @@ namespace
     size_t case_failed = 0;
     size_t case_failed_before = 0;
 
-    handlers_t defaults = default_handlers;
-    handlers_t handlers = defaults;
+    handlers_t& defaults() {
```

Comment:
`handlers_t` is not a class, it is a POD. 
The cause of the trouble is related to the initialization of these handlers: the initialization is made by copy of  default_handlers. 
This is **not** something done at compile time, the initialization happen at **runtime**. 
That mean that the variables `defaults` and `handlers` will be kept even if there is no code using them after the initialization. 
That also means that the variable `default_handlers` will be kept and the various function pointers which initialize it will be kept. 

It is not possible to use directly a `SingletonPtr` here because there is no way to specify the initialization sequence. 
On the other hand, it can be worked around by creating a new class  derived from handlers_t which default initialize to `default_handlers` . Do you prefer this solution ?